src/sleep/sleep.c

```diff
@@ -286,6 +286,11 @@ static int execute_s2h(const SleepConfig *sleep_config) {
                 last_capacity = r;
                 before_timestamp = now(CLOCK_BOOTTIME);
 
+                previous_discharge_rate = get_battery_discharge_rate();
```

Comment:
I'd also assign to `r` here since the function could return an error

Reply:
yes. will make that a habit now.